src/agents/intent_agent_graph.py

```diff
@@ -11,71 +11,97 @@
 from chains.question_lookup_chain import QuestionLookupChain
 from chains.kg_chain import KGChain
 import config as app_config
+from typing import Sequence, TypedDict, Annotated, Any
+import config
 
 
 class AgentState(TypedDict):
-    # The agent state is the input to each node in the graph
-    # Our state Schema (https://langchain-ai.github.io/langgraph/concepts/low_level/#schema)
-    # The annotation tells the graph that new messages will always
-    # be added to the current states
     input: Annotated[Sequence[BaseMessage], operator.add]
-    # The 'next' field indicates where to route to next
-    next: str
-    chat_history: List = []
+    # Stores a list of the next agents to process the request
+    next: Annotated[List[str], operator.add] 
```

Comment:
Should this be just a list with out the `operator.add` . Cause it might be the case when using the add operator , it will keep appending the list of next. So for eg , after supervisor finishes, let's say supervisor says next is `QV_lookup` and returns  ["QV_Lookup"] , i think the langgraph runtime will see ["Supervisor", "QV_Lookup"] after supervisor is called instead of just the qv_lookup ... wonder if that is the case ?